src/backend/routers/user.py

```diff
@@ -88,10 +90,15 @@ async def login(
     password = user_login.password
 
     user: User | None = await User.prisma().find_first(where={"username": username})
-
-    if user is None or (not await check_password(password, user.password)):
+    if user is None:
+        raise HTTPException(status_code=404, detail="User does not exist")
+    db_pw = await Password.prisma().find_first(where={"user_id": user.id})
+    # this case should NEVER happen, but we include it here for type-safety
+    if db_pw is None:
+        raise HTTPException(status_code=404, detail="User exists but does not have a password")
```

Comment:
It'd be wise to add a `logger.error()` here-  it will be useful if we plan to implement a monitoring tool or a similar solution in the future.

Reply:
Again, this case will never happen but :shrug: .

Reply:
Resolved [aefbc65](https://github.com/idkbrowby/gamingrealm-backend/pull/31/commits/aefbc6508dc44a74a9ed8af29ef31930472a9de3)